### ISSUE TITLE
AnimationGroup and Succession not working properly with integer lag_ratio

### ISSUE THREAD

**Issue by PhilippImhof** [ManimCommunity/manim#888]
## Description of bug / unexpected behavior

Consider the following code:

```python
class Test(Scene):
    def construct(self):
        self.play(
            AnimationGroup(
               *map(FadeIn,[Text("a").shift(LEFT),Text("b").shift(RIGHT)]),
              lag_ratio=1)
        )
        self.wait()
```

It shows `b`, fades in `a` and then removes `b` before fading it in.

## Expected behavior

It should fade in the text `a`and, once this is done, fade in the text `b`.

## Additional comments

It seems to be a problem with the fact that `lag_ratio` is an integer. If I set `lag_ratio=1.0`, the code works as expected. I will prepare a PR to change `Succession` accordingly, but it might be interesting to know why this happens.

**Comment by leotrs** [comment]
I think @behackl was looking into a similar problem a few weeks ago.

However, I have no idea why this would happen for integers but not for floats...

**Comment by behackl** [comment]
This was fixed at some point, cannot reproduce with `v0.8.0`.
